### ISSUE TITLE
Copy-out on exit signal

### ISSUE THREAD

**Issue by QinshiWang** [verified-network-toolchain/petr4#321]
Out parameters are still copied out on exit signal. Is that okay?

**Comment by QinshiWang** [comment]
I know that when the signal is exit, the stack frame (`mem`) is irrelevant, but it sounds to be error-prone.

**Comment by jnfoster** [comment]
Yes, this is the desired behavior. I can dig up the spec issue if you're curious...
